### PR TITLE
comment for jumptable

### PR DESCRIPTION
**Pull request by chenzhitong**

### CODE REVIEW

File: src/Neo.VM/JumpTable/JumpTable.Push.cs [hunk 1]
```diff
@@ -16,56 +16,121 @@
...
             engine.Push(new BigInteger(instruction.Operand.Span));
         }
 
+        /// <summary>
+        /// Pushes an 16-bit signed integer onto the evaluation stack.
+        /// </summary>
+        /// <param name="engine">The execution engine.</param>
+        /// <param name="instruction">The instruction being executed.</param>
+        /// <summary>
+        /// <see cref="OpCode.PUSHINT16"/>
```
Comment (shargon) [inline src/Neo.VM/JumpTable/JumpTable.Push.cs:42]: see goes inside summary? we have a warning

Comment (chenzhitong) [reply src/Neo.VM/JumpTable/JumpTable.Push.cs:42]: modified

File: src/Neo.VM/JumpTable/JumpTable.Bitwisee.cs
Comment (cschuchardt88) [inline src/Neo.VM/JumpTable/JumpTable.Bitwisee.cs:1]: Add `CompareTo` method.

File: src/Neo.VM/JumpTable/JumpTable.Compound.cs [hunk 2]
```diff
@@ -173,13 +242,21 @@ public virtual void Size(ExecutionEngine engine, Instruction instruction)
...
         {
             var key = engine.Pop<PrimitiveType>();
             var x = engine.Pop();
+            // Check the type of the top item and perform the corresponding action.
             switch (x)
             {
+                // For arrays, check if the index is within bounds and push the result onto the stack.
                 case VMArray array:
                     {
                         var index = (int)key.GetInteger();
```
Comment (cschuchardt88) [inline src/Neo.VM/JumpTable/JumpTable.Compound.cs:262]: Overflow and underflow checking needs to be done.

File: src/Neo.VM/JumpTable/JumpTable.Compound.cs [hunk 3]
```diff
@@ -188,11 +265,13 @@ public virtual void HasKey(ExecutionEngine engine, Instruction instruction)
...
+                // For maps, check if the key exists and push the result onto the stack.
                 case Map map:
                     {
                         engine.Push(map.ContainsKey(key));
                         break;
                     }
+                // For buffers, check if the index is within bounds and push the result onto the stack.
                 case Types.Buffer buffer:
                     {
                         var index = (int)key.GetInteger();
```
Comment (cschuchardt88) [inline src/Neo.VM/JumpTable/JumpTable.Compound.cs:277]: Overflow and underflow checks need to be added and should of been.

File: src/Neo.VM/JumpTable/JumpTable.Compound.cs [hunk 4]
```diff
@@ -201,6 +280,7 @@ public virtual void HasKey(ExecutionEngine engine, Instruction instruction)
                         engine.Push(index < buffer.Size);
                         break;
                     }
+                // For byte strings, check if the index is within bounds and push the result onto the stack.
                 case ByteString array:
                     {
                         var index = (int)key.GetInteger();
```
Comment (cschuchardt88) [inline src/Neo.VM/JumpTable/JumpTable.Compound.cs:286]: Overflow and underflow need to be done.

File: src/Neo.VM/JumpTable/JumpTable.Compound.cs [hunk 5]
```diff
@@ -91,12 +118,24 @@ public virtual void Unpack(ExecutionEngine engine, Instruction instruction)
...
         }
 
+        /// <summary>
+        /// Creates a new empty array with zero elements on the evaluation stack.
+        /// <see cref="OpCode.NEWARRAY0"/>
+        /// </summary>
+        /// <param name="engine">The execution engine.</param>
+        /// <param name="instruction">The instruction being executed.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public virtual void NewArray0(ExecutionEngine engine, Instruction instruction)
```
Comment (cschuchardt88) [inline src/Neo.VM/JumpTable/JumpTable.Compound.cs:128]: Change to `NewNullArray` method or add it?

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]
You to change all method names to have a `Prefix` of `Push` or `Pop` or `Add` or `Remove`. So we can tell what's going on.

**Comment by chenzhitong** [comment]
> @chenzhitong can you also add the satck change to the comments? like pop two, push one?

OK

File: src/Neo.VM/JumpTable/JumpTable.Compound.cs (hunk 5 above)
Comment (Jim8y) [reply src/Neo.VM/JumpTable/JumpTable.Compound.cs:131]: This is the name of the opcode.

File: src/Neo.VM/JumpTable/JumpTable.Compound.cs (hunk 4 above)
Comment (Jim8y) [reply src/Neo.VM/JumpTable/JumpTable.Compound.cs:292]: this is comments, logic issues should be in another pr.

File: src/Neo.VM/JumpTable/JumpTable.Bitwisee.cs
Comment (Jim8y) [reply src/Neo.VM/JumpTable/JumpTable.Bitwisee.cs:1]: This pr only focus on adding comments to existing codes.

File: src/Neo.VM/JumpTable/JumpTable.Push.cs [hunk 6]
```diff
@@ -16,56 +16,121 @@
...
```
Comment (shargon) [inline src/Neo.VM/JumpTable/JumpTable.Push.cs:42]: more "see" outside more

File: src/Neo.VM/JumpTable/JumpTable.Bitwisee.cs
Comment (shargon) [reply src/Neo.VM/JumpTable/JumpTable.Bitwisee.cs:1]: > Add `CompareTo` method.

Focus on the changes in the pull request.

**Review by shargon — APPROVED** [review]

**Comment by shargon** [comment]
> You to change all method names to have a `Prefix` of `Push` or `Pop` or `Add` or `Remove`. So we can tell what's going on.

Review not related to the PR. I will merge it.

**Pull request merged**